tests/Aspire.Workload.Tests/PerTestFrameworkTemplatesTests.cs

```diff
@@ -79,7 +79,7 @@ await using (var context = await CreateNewBrowserContextAsync())
         res = await cmd.ExecuteAsync($"test -c {config}");
 
         Assert.True(res.ExitCode != 0, $"Expected the tests project build to fail");
-        Assert.Matches("System.ArgumentException.*Resource 'webfrontend' not found.", res.Output);
+        Assert.Matches("System.InvalidOperationException.*Sequence contains no matching element", res.Output);
```

Comment:
1. (In the existing code) @radical - Why are we expecting the tests to fail?
2. For the updated code, isn't the original exception better? Seems like we are making this worse.

Reply:
> 1. (In the existing code) @radical - Why are we expecting the tests to fail?

The test creates an `aspire` project, then a `aspire-xunit` project, uncomments the code in the generated test file, and runs that. And that causes the test to fail because the test is expecting a `webfrontend` to exist. So, this what the user would see.

We could add a `webfrontend` to the solution, and patch that up so it gets used in this test. Do we want to do that?

Reply:
We could move the `CreateHttpClient` call to before the `WaitForResourceAsync` call to retain the original exception message and not bother with finding the resource in the `DistributedApplicationModel`.

Reply:
I went ahead and adjusted the order of lines in the test templates so that the original exception is thrown. @radical I do think it would be better to update these tests to add a resource though so the tests actually pass. Perhaps we could just base it on the starter template instead (without the tests project option) in order to get the web project by default?

Reply:
I was under the understanding we were validating the scenario @radical describes above. What is the experience when a dev adds the test project, and just runs it. Do they get an understandable error?

Reply:
If you just add the test project and run it you don't get an error. The test here actually uncomments the sample code and then runs the test, which fails because the sample code assumes there's a resource to test against (which the comments say).